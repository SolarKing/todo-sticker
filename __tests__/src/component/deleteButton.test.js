// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import { Modal } from 'antd';
import DeleteTodoButton from '../../../src/component/deleteButton/deleteButton';

jest.mock('antd', () => {
    let Modal = {
        confirm: jest.fn(),
    };
    return {
        Modal
    };
});

describe('DeleteTodoButton', () => {
    let wrapper: ShallowWrapper;
    let handleDeleteTodo = jest.fn();

    beforeAll(() => {
        wrapper = shallow(<DeleteTodoButton handleDeleteTodo={handleDeleteTodo} todoId={10} />);
    });

    it('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should excute modal confirm when clicked', () => {
        wrapper.find('a').simulate('click');
        expect(Modal.confirm.mock.calls.length).toEqual(1);
    });

    it('should execute handleDeleteTodo when confirming successfully', () => {
        wrapper.find('a').simulate('click');
        Modal.confirm.mock.calls[0][0].onOk();
        expect(handleDeleteTodo.mock.calls.length).toEqual(1);
        expect(handleDeleteTodo.mock.calls[0][0]).toEqual(10);
    });

});