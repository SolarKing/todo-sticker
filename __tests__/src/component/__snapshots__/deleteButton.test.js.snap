// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteTodoButton should render correct 1`] = `
<a
  onClick={[Function]}
>
  Delete
</a>
`;
